lib/proxi/local/sdrf_handler.py

```
@@ -0,0 +1,729 @@
+                    if first_line:
+                        sdrf_data['titles'] = line.strip().split(filetype['delimiter'])
+                        first_line = False
+                        continue
+                    columns = line.strip().split(filetype['delimiter'])
+                    #### Add empty values at the end if the row has fewer columns of data than the titles
+                    while len(columns) < len(sdrf_data['titles']):
+                        columns.append('')
+                    sdrf_data['rows'].append(columns)
+
+        elif filetype['type'] == 'xls':
+            source_sdrf_data = pandas.read_excel(filepath)
```

Ruff (0.15.12)

[error] Undefined name `pandas`

(F821)